pkg/cache/node_info.go

```diff
@@ -109,6 +112,40 @@ func (ni *NodeInfo) GetCapacity() *resources.Resource {
 	return ni.totalResource.Clone()
 }
 
+func (ni *NodeInfo) SetCapacity(newCapacity *resources.Resource) {
+	ni.lock.Lock()
+	defer ni.lock.Unlock()
+	if resources.Equals(ni.totalResource, newCapacity) {
+		log.Logger().Info("skip updating capacity, not changed")
+		return
+	}
+
+	ni.totalResource = newCapacity
+	ni.availableResource = ni.totalResource.Clone()
+	ni.availableResource.SubFrom(ni.allocatedResource)
+	ni.availableResource.SubFrom(ni.occupiedResource)
```

Comment:
Move to a separate function `setAvailabaleResource()` and call from:
* SetCapacity()
* SetOccupiedResource()
* NewNodeInfo()

Keeps the logic the same everywhere.

Reply:
Make sense to me.
I moved to code to:

```
// refresh node available resource based on the latest allocated and occupied resources.
// this call assumes the caller already acquires the lock.
func (ni *NodeInfo) refreshAvailableResource() {
	ni.availableResource = ni.totalResource.Clone()
	ni.availableResource.SubFrom(ni.allocatedResource)
	ni.availableResource.SubFrom(ni.occupiedResource)
}
```